test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightOutputContextTests.cs

```diff
@@ -438,14 +438,23 @@ public async Task WriteBatchAsync()
                     var operationRequestMessage = await batchWriter.CreateOperationRequestMessageAsync(
                         "POST", new Uri($"{ServiceUri}/Orders"), "1");
 
-                    using (var messageWriter = new ODataMessageWriter(operationRequestMessage))
+                    var messageWriter = new ODataMessageWriter(operationRequestMessage);

```

Comment:
Could you have refactored this in the following manner:
```csharp
#if NETCOREAPP3_1_OR_GREATER
                    await using (var messageWriter = new ODataMessageWriter(operationRequestMessage))
#else
                    using (var messageWriter = new ODataMessageWriter(operationRequestMessage))
#endif
                    {
                        var resourceWriter = await messageWriter.CreateODataResourceWriterAsync(this.orderEntitySet, this.orderEntityType);
                        var orderResource = CreateOrderResource();
                        await resourceWriter.WriteStartAsync(orderResource);
                        await resourceWriter.WriteEndAsync();
                    }
```

I'd be curious whether if you did it this way `await messageWriter.DisposeAsync();` would automatically get triggered for `NETCOREAPP3_1_OR_GREATER`; otherwise `messageWriter.Dispose()`

Reply:
I actually used this pattern in later code, I just hadn't thought of it until I was half-way through the refactoring.

Reply:
I refactored the remaining `try-finally` versions.